packages/expo-cli/src/commands/webhooks.ts

```diff
@@ -1,150 +1,171 @@
-    );
-  }
+async function addAsync(
+  projectRoot: string,
+  { url, event, ...options }: { url?: string; event?: WebhookEvent; secret?: string }
+) {
+  invariant(typeof url === 'string' && /^https?/.test(url), '--url: a HTTP URL is required');
+  invariant(typeof event === 'string', '--event: string is required');
+  const secret = validateSecret(options) || generateSecret();
 
-  if (secret) {
-    const secretString = String(secret);
-    if (secretString.length < 16 || secretString.length > 1000) {
-      throw new Error('Webhook secret has be at least 16 and not more than 1000 characters long');
-    }
-  } else {
-    secret = await _askForSecret();
-  }
+  const { experienceName, project, client } = await setupAsync(projectRoot);
 
-  return { url, secret, event };
+  const spinner = ora(`Adding webhook to ${experienceName}`).start();
+  await client.postAsync(`projects/${project.id}/webhooks`, { url, event, secret });
+  spinner.succeed();
 }
 
-function _sanitizeEvent(event: string | undefined, required = false): string | undefined {
-  if (!event) {
-    // we don't have anything to sanitize here, continue
-    return event;
-  }
+async function updateAsync(
+  projectRoot: string,
+  {
+    id,
+    url,
+    event,
+    ...options
+  }: { id?: string; url?: string; event?: WebhookEvent; secret?: string }
+) {
+  invariant(typeof id === 'string', '--id must be a webhook ID');
+  invariant(event == null || typeof event === 'string', '--event: string is required');
+  let secret = validateSecret(options);
+
+  const { project, client } = await setupAsync(projectRoot);
+
+  const webhook = await client.getAsync(`projects/${project.id}/webhooks/${id}`);
+  event = event ?? webhook.event;
+  secret = secret ?? webhook.secret;
+
+  const spinner = ora(`Updating webhook ${id}`).start();
+  await client.patchAsync(`projects/${project.id}/webhooks/${id}`, { url, event, secret });
+  spinner.succeed();
+}
 
-  if (!WEBHOOK_TYPES.includes(event)) {
-    throw new Error(`Unsupported webhook type: ${event}`);
-  }
+async function removeAsync(projectRoot: string, { id }: { id?: string }) {
+  invariant(typeof id === 'string', '--id must be a webhook ID');
+  const { project, client } = await setupAsync(projectRoot);
 
-  return event;
+  await client.deleteAsync(`projects/${project.id}/webhooks/${id}`);
 }
 
-async function _askForSecret(): Promise<string> {
-  const { secret } = await inquirer.prompt({
-    type: 'password',
-    name: 'secret',
-    message: 'Webhook secret (at least 16 and not more than 1000 characters):',
-  });
-  if (secret.length < 16 || secret.length > 1000) {
-    log.error('Webhook secret has be at least 16 and not more than 1000 characters long');
-    return await _askForSecret();
-  } else {
+function validateSecret({ secret }: { secret?: string }): string | null {
+  if (secret) {
+    invariant(
+      secret.length >= SECRET_MIN_LENGTH && secret.length < SECRET_MAX_LENGTH,
```

Comment:
If we say the secret needs to be between 16 and 1000 characters long, I could see people trying 1000 characters and expecting it to work:
```suggestion
      secret.length >= SECRET_MIN_LENGTH && secret.length <= SECRET_MAX_LENGTH,
```